Lab 9: initial draft of conversion to Markdown

joshmhanson opened the pull request.

joshmhanson committed:
Lab 9: initial draft of conversion to Markdown

joshmhanson committed:
Lab 9: Switch to Markdown tables when possible

Allows using inline formatting

hXtreme commented on _labs/lab09.markdown:
```diff
@@ -1,6 +1,363 @@
+/lab9$ mkdir build && cd build
+/lab9/build$ cmake ..
+/lab9/build$ make
+/lab9/build$ export LD_LIBRARY_PATH=/lab9/build:$LD_LIBRARY_PATH
```
Remove this

hXtreme commented on _labs/lab09.markdown:
````diff
@@ -1,6 +1,363 @@
+/lab9/build$ export LD_LIBRARY_PATH=/lab9/build:$LD_LIBRARY_PATH
+```
+
+**The `export LD_LIBRARY_PATH` command should be run on each terminal session you begin.**
````
Remove this

hXtreme commented on _labs/lab09.markdown:
```diff
@@ -1,6 +1,363 @@
+You will need to encode a C program into this symbolic interpreter API as well as write the code that drives the dynamic symbolic execution.
+We provide several details on how to do this in the following sections.
+
+This lab assumes that input programs only have integer variables (no pointers or other types of variables) and do not have functions (no `CallInstr`).
```
typo `CallInst`

hXtreme commented on _labs/lab09.markdown:
```diff
@@ -1,6 +1,363 @@
+#### Understanding Z3
+
+Z3 is a theorem prover developed at Microsoft.
+It's a large and complex tool, so this will serve as a cursory guide for its capabilities and what it can do. Consider a simple, generic system of equations such as the following, where X and Y are integers:
```
Break into multiple lines.

hXtreme replied:
Add backtick quotes to `X` and `Y`

hXtreme commented on _labs/lab09.markdown:
```diff
@@ -1,6 +1,363 @@
+The first component of this dynamic symbolic execution implementation is instrumentation of the input program, which is done in `src/Instrument.cpp`.
+This follows the familiar format and pattern seen in prior labs, except now this LLVM pass will inject various functions defined in `src/Runtime.cpp`, accompanied with the appropriate metadata from each valid LLVM Instruction.
+This will enable DSE to interact with Z3 at runtime.
+Specifically, these are the functions that will require instrumentation (from `include/Instrument.h`):
```
Rename all occurrances of `include/Instrument.h` to `include/DSEInstrument.h` similarly for `src/Instrument.cpp`

hXtreme commented on _labs/lab09.markdown:
```diff
@@ -1,6 +1,363 @@
+This is because most of these programming languages are imperatively-directed, meaning there's a sequence of commands needed to solve the problem.
+
+On the other hand, Z3 has a declarative interface, which in this case means all you need to give it is the list of constraints (in this case, `X < Y` and `X > 2`).
+Plug the following into an [online Z3 solver](https://rise4fun.com/Z3/tutorial/guide) to see the results:
```
Switch to https://compsys-tools.ens-lyon.fr/z3/index.php

rise4fun is down

joshmhanson committed:
Lab 9 document: minor updates

- make changes to match the code updates to Lab 9
- fix broken link and a typo
- minor formatting fixes
